server/channel/src/AIManager.cpp

```
@@ -1010,17 +1018,21 @@ bool AIManager::UpdateState(const std::shared_ptr<ActiveEntityState>& eState,
     }
 
     if(aiState->IsIdle() &&
-        !aiState->ActionOverridesKeyExists("idle"))
+        !aiState->ActionOverridesKeyExists("idle") &&
+        !aiState->HasFollowTarget() &&
+        !aiState->GetCurrentCommand())
     {
+        // Nothing to do
         return false;
```

What is the difference between "nothing to do" and an actual error condition if they both silently return false?

---

Not sure why there would be an error condition here. This is identical to an intentional "do not act" for an idle entity.

---

Sorry meant the condition above. See that comment.